cleaner.py

```diff
@@ -1,85 +1,225 @@
+shortNameLookup = df({'shortName': shortNames, 'longName': longNames})
+regularSeason['favorite'] = regularSeason['Fav_Team'].transform(lambda name: shortNameLookup[shortNameLookup['longName'] == name]['shortName'])
+
+regularSeason['HomeFav'] = (regularSeason['favorite'] == '000').map({True: 0, False: -1}) 
+regularSeason.loc[regularSeason['favorite'] == regularSeason['Home_Team'], 'HomeFav'] = 1
+regularSeason['underdog'] = (regularSeason['favorite'] == regularSeason['team']).map({True: 0, False: 1})
+
+regularSeason.fillna(0)
+
+regularSeason['Home_Vegas_Spread'] = regularSeason[regularSeason['HomeFav'] == 1]['Fav_Spread']
+regularSeason['Home_Vegas_Spread'] = np.where(regularSeason['HomeFav'] == 0, 0, regularSeason['Fav_Spread'].transform(lambda fav: fav * -1))
+regularSeason['Home_Actual_Spread'] = -1 * (regularSeason['Home_Score'] - regularSeason['Away_Score'])
+
+away_pass_stats = regularSeason[['aCmp', 'aAtt', 'aYd', 'aTD', 'aINT']]
+home_pass_stats = regularSeason[['hCmp', 'hAtt', 'hYd', 'hTD', 'hINT']]
+
+away_rush_stats = regularSeason[['aRush', 'aRYds', 'aRTDs']]
+home_rush_stats = regularSeason[['hRush', 'hRYds', 'hRTDs']]
+away_pen_yds = regularSeason[['aPen', 'aPenYds']]
+home_pen_yds = regularSeason[['hPen', 'hPenYds']]
+away_third_downs = regularSeason[['aThrdConv', 'aThrd']]
+home_third_downs = regularSeason[['hThrdConv', 'hThrd']]
+away_fourth_downs = regularSeason[['aFrthConv', 'aFrth']]
+home_fourth_downs = regularSeason[['hFrthConv', 'hFrth']]
+
+@staticmethod
+def assignStats(regularSeason: df, column: str):
+    offensive = column
+    defensive = f'd{column}'
+    homestat = f'h{column}'
+    awaystat = f'a{column}'
+    regularSeason[offensive] = np.where(regularSeason['at'] == '@', regularSeason[awaystat], regularSeason[homestat])
+    regularSeason[defensive] = np.where(regularSeason['at'] == '@', regularSeason[homestat], regularSeason[awaystat])
+
+passing = ['Cmp', 'Att', 'Yd', 'TD', 'INT']
+rushing = ['Rush', 'RYds', 'RTDs']
+downs = ['ThrdConv', 'Thrds', 'FrthConv', 'Frths']
+
+for column in passing:
+    assignStats(regularSeason, column)
+regularSeason['team_pass_eff'] = regularSeason['Yd'] / regularSeason['Att']
+regularSeason['team_pass_def'] = regularSeason['dYd'] / regularSeason['dAtt']
+
+for column in rushing:
+    assignStats(regularSeason, column)
+regularSeason['team_rush_eff'] = regularSeason['RYds'] / regularSeason['Rush']
+regularSeason['team_rush_def'] = regularSeason['dRYds'] / regularSeason['dRush']
+
+for column in downs:
+    assignStats(regularSeason, column)
+regularSeason['third_eff'] = regularSeason['ThrdConv'] / regularSeason['Thrds']
+regularSeason['third_def'] = regularSeason['dThrdConv'] / regularSeason['dThrds']
+regularSeason['fourth_eff'] = regularSeason['FrthConv'] / regularSeason['Frths']
+regularSeason['fourth_def'] = regularSeason['dFrthConv'] / regularSeason['dFrths']
+
+regularSeason['fourth_eff'] = regularSeason['fourth_eff'].fillna(0.0001)
```

Comment:
could we add a comment here to follow up on the proper way to do this replace? I used 0,0001 originally but that's not the right way to do it

Reply:
added this is an issue rather than stopping merge to make the change